beacon/goclient/goclient.go

```diff
@@ -218,8 +217,9 @@ func New(
 		zap.String("version", client.nodeVersion),
 	)
 
-	// Start registration submitter.
-	go client.registrationSubmitter(slotTickerProvider)
+	if operatorDataStore != nil {
+		go client.registrationSubmitter(slotTickerProvider)
+	}
```

Comment:
why would it be nil?  and if it will be nil at the point of creation but then it will be created then `registrationSubmitter` won't ever start..

Reply:
At the moment in `mockClient` in tests, but maybe it will be changed

Reply:
The operator ID used to be 0 in `mockClient`, and now the client waits until operator ID is not 0, so it's a workaround for tests

Reply:
Although I think it can be set to any non-zero value in tests instead